5 add contactus page

androidmini9x opened the pull request:
- Create Contact Us page
- Add Test to the form
- Add Emailing library

androidmini9x committed:
Add ContactUs Page
Relates #5

androidmini9x committed:
Add: emailJs library and implement it to the form

Iseluin commented on src/App.jsx:
```diff
@@ -1,55 +1,10 @@
-import logo from './logo.svg';
-import { Counter } from './features/counter/Counter';
 import './App.css';
+import Contactus from './containers/Contactus';
```
The folder name should be "pages" not "containers". It is not wrong to call it containers at all but "pages" is the naming convention we go with.

Iseluin commented on src/containers/Contactus.jsx:
```diff
@@ -0,0 +1,230 @@
+import { Formik, Form, Field, ErrorMessage } from 'formik';
+import { useRef } from 'react';
```
Please leave a line between local imports and other imports.

Iseluin commented on src/containers/Contactus.jsx:
```diff
@@ -0,0 +1,230 @@
+                errors.name = 'Name is required';
+              }
+              if (values.message === '') {
+                errors.message = 'Bodys Message is required';
```
Let's change the error message for the message input to something like "Please add a message at least one character long".

Iseluin commented on src/containers/Contactus.jsx:
```diff
@@ -0,0 +1,230 @@
+            </h3>
+            <div className="mt-2 px-7 py-3">
+              <p className="text-sm text-gray-500">
+                Your message has not been sent successfully.
```
Can we change this error message to something like "A problem has occured and we couldn't deliver your mail"

Iseluin commented on src/containers/Contactus.jsx:
```diff
@@ -0,0 +1,230 @@
+  );
+}
+
+// const modal = document.getElementById('my-modal');
+// const btn = document.getElementById('open-btn');
+// const button = document.getElementById('ok-btn');
+// btn.onclick = function () {
+//   modal.style.display = "block";
+// }
+// button.onclick = function () {
+//   modal.style.display = "none";
+// }
+// window.onclick = function (event) {
+//   if (event.target == modal) {
+//     modal.style.display = "none";
+//   }
+// }
```
Delete the unnecessary comments please.

Iseluin commented on src/containers/Contactus.jsx:
```diff
@@ -0,0 +1,230 @@
+function Contactus() {
+  const modelRef = useRef();
+
+  const sendEmail = (values) => {
```
Can you try to move the sendEmail function to a src/utils/utils.js file and import it here ? That way the code would look cleaner. If it's not possible since the function has lots of inbuilt emailJS parameters inside, just let me know.

Iseluin commented on src/__tests__/Contactus.test.js:
```diff
@@ -0,0 +1,50 @@
+});
+
+test('Check entering in the inputs fields', async () => {
+  // fireEvent.change(screen.queryByLabelText(/full name/i), {
+  //     target: { value: '' }
+  // });
```
Remove the comments please.

Iseluin commented on src/containers/Contactus.jsx:
```diff
@@ -0,0 +1,230 @@
+
+  return (
+    <section className="text-gray-600 body-font relative">
+      <h1 className="text-gray-400 text-5xl p-5 font-medium">CONTACT US.</h1>
```
Remove the dot at the end.

Iseluin reviewed (changes requested):
Great work @androidmini9x ! This looks really good. I have some feedback for you on the comments below, please adjust your code accordingly. Other than that, for this PR:

1- There are lots of eslint errors in the page. Please resolve them WITHOUT adding exceptions.

2- It seems like emailJS works but just to be sure can you check if it sent you my test mail? 

3-  There is a responsivity issue when the screen size is < 1024, illustration shrinks too much. Please fix that.

4- Since we did not have a figma design for this page, you used your creativity. Please adjust the color scheme of our figma design to the page and the illustration.

5- Please use PascalCase for the naming of the files. Ex: ContactUs.jsx, ContactUs.test.js etc.

androidmini9x committed:
Fix: fix eslint errors and restructure file&folder

androidmini9x commented:
- I fixed the eslint errors that related to my component.
- The emailJS works and I got your message.
- I fixed the responsivity issue.
- I renamed the folder to pages, and the file to PascalCase pattern.
- I removed unnecessary comments and codes.

Iseluin commented on src/App.jsx:
```diff
@@ -1,55 +1,10 @@
-import logo from './logo.svg';
-import { Counter } from './features/counter/Counter';
 import './App.css';
+import ContactUs from './pages/ContactUs';
```
lets move it to src/pages/ContactUs/<your-filename>.

Iseluin commented on src/__tests__/ContactUs.test.js:
```diff
@@ -0,0 +1,48 @@
+
+  await wait(() => {
+    fireEvent.click(screen.queryByText(/send message/i));
+  });
+
+  expect(screen.getByText(/name is required/i)).toBeInTheDocument();
+  expect(screen.getByText(/email is required/i)).toBeInTheDocument();
+  expect(screen.getByText(/bodys Message is required/i)).toBeInTheDocument();
+
+  await wait(() => {
+    fireEvent.change(screen.queryByLabelText(/full name/i), {
+      target: { value: 'Ahmed' },
+    });
+    fireEvent.change(screen.queryByLabelText(/email/i), {
+      target: { value: 'ahmed' },
+    });
+    fireEvent.change(screen.queryByLabelText(/message/i), {
+      target: { value: 'Hello, World' },
+    });
+  });
+
+  expect(screen.getByText(/invalid email address format/i)).toBeInTheDocument();
+
+  await wait(() => {
+    fireEvent.change(screen.queryByLabelText(/full name/i), {
+      target: { value: 'Ahmed' },
+    });
+    fireEvent.change(screen.queryByLabelText(/email/i), {
+      target: { value: 'ahmed@example.com' },
+    });
+    fireEvent.change(screen.queryByLabelText(/message/i), {
+      target: { value: 'Hello, World' },
+    });
+    fireEvent.click(screen.queryByText(/send message/i));
+  });
+
+  expect(screen.queryByLabelText(/full name/i).value).toBe('Ahmed');
+  expect(screen.queryByLabelText(/email/i).value).toBe('ahmed@example.com');
+  expect(screen.queryByLabelText(/message/i).value).toBe('Hello, World');
+});
```
Some tests here fails when npm run test is run. Can you check them out ?

Iseluin commented on src/assets/contactus.svg:
```diff
@@ -0,0 +1,239 @@
+              <path d="M924.37,509.47a12.32,12.32,0,0,1-11.71-17.56,9.11,9.11,0,0,1,2.48,2.47,39.47,39.47,0,0,0,2.44,4.07c1.91,2.26,5.19,3,8.16,2.86s5.84-1,8.8-1.25c.41,0,.83-.07,1.24-.08A12.31,12.31,0,0,1,924.37,509.47Z" transform="translate(-52 -162.63)" opacity="0.1"></path>
+              <path d="M894.49,529.35A12.3,12.3,0,0,1,882.68,512a10.15,10.15,0,0,1,2.6,2.5c1,1.35,1.55,2.91,2.67,4.15,2.08,2.32,5.57,3.13,8.72,3.06s6-.82,9-1.05A12.31,12.31,0,0,1,894.49,529.35Z" transform="translate(-52 -162.63)" opacity="0.1"></path>
+              <path d="M900.57,553.26a12.32,12.32,0,0,1-22.45-10,12.06,12.06,0,0,1,2.7,2.41c1.18,1.43,1.95,3,3.3,4.38,2.52,2.47,6.58,3.48,10.2,3.58A53.94,53.94,0,0,0,900.57,553.26Z" transform="translate(-52 -162.63)" opacity="0.1"></path>
+              <path d="M1047.39,729c2.61-4.83-.35-10.76-3.76-15.07s-7.55-8.79-7.46-14.29c.12-7.89,8.5-12.55,15.2-16.74a73.9,73.9,0,0,0,13.64-11,19.93,19.93,0,0,0,4.2-5.61c1.38-3.09,1.34-6.6,1.26-10q-.43-16.9-1.67-33.76" transform="translate(-52 -162.63)" fill="none" stroke="#3f3d56" stroke-miterlimit="10" stroke-width="4"></path>
+              <path d="M1081.52,621.93a12.29,12.29,0,0,0-6.14-10.08l-2.76,5.45.08-6.6a12.08,12.08,0,0,0-4-.49,12.31,12.31,0,1,0,12.87,11.72Z" transform="translate(-52 -162.63)" fill="#667eea"></path>
+              <path d="M1059.57,705.26a12.31,12.31,0,1,1,.59-9.92l-7.69,6.26,8.46-2A12.24,12.24,0,0,1,1059.57,705.26Z" transform="translate(-52 -162.63)" fill="#667eea"></path>
+              <path d="M1053.49,681.35a12.31,12.31,0,0,1-3.91-24.15l-.06,5.07,2.79-5.51h0a12.32,12.32,0,0,1,12.87,11.73,12.3,12.3,0,0,1-11.72,12.87Z" transform="translate(-52 -162.63)" fill="#667eea"></path>
+              <path d="M1083.37,661.47a12.31,12.31,0,1,1,5.45-23l-2.18,6,4.48-4.3a12.24,12.24,0,0,1,4,8.5,11.88,11.88,0,0,1-.31,3.39A12.31,12.31,0,0,1,1083.37,661.47Z" transform="translate(-52 -162.63)" fill="#667eea"></path>
+              <path d="M1079.87,625.21c-2.84.32-5.61,1.2-8.46,1.38s-6-.51-7.79-2.73a38.2,38.2,0,0,1-2.27-4,8.85,8.85,0,0,0-3.1-2.92,12.31,12.31,0,1,0,23,8.17C1080.79,625.13,1080.33,625.16,1079.87,625.21Z" transform="translate(-52 -162.63)" opacity="0.1"></path>
+              <path d="M1083.37,661.47a12.32,12.32,0,0,1-11.71-17.56,9.11,9.11,0,0,1,2.48,2.47,39.47,39.47,0,0,0,2.44,4.07c1.91,2.26,5.19,3,8.16,2.86s5.84-1,8.8-1.25c.41,0,.83-.07,1.24-.08A12.31,12.31,0,0,1,1083.37,661.47Z" transform="translate(-52 -162.63)" opacity="0.1"></path>
+              <path d="M1053.49,681.35A12.3,12.3,0,0,1,1041.68,664a10.15,10.15,0,0,1,2.6,2.5c1,1.35,1.55,2.91,2.67,4.15,2.08,2.32,5.57,3.13,8.72,3.06s6-.82,9-1.05A12.31,12.31,0,0,1,1053.49,681.35Z" transform="translate(-52 -162.63)" opacity="0.1"></path>
+              <path d="M1059.57,705.26a12.32,12.32,0,0,1-22.45-10,12.06,12.06,0,0,1,2.7,2.41c1.18,1.43,2,3,3.3,4.38,2.52,2.47,6.58,3.48,10.2,3.58A53.94,53.94,0,0,0,1059.57,705.26Z" transform="translate(-52 -162.63)" opacity="0.1"></path>
+              <path d="M530,587.91a9.81,9.81,0,0,0-3.91-7.92,11.5,11.5,0,0,0-9.61-11.28,11.48,11.48,0,0,0-7.65-8.55h0a9.76,9.76,0,0,0-1.42-.33l-.14,0-.25,0-.2,0-.22,0-.27,0h-.61a11.26,11.26,0,0,0-10.88,9.62c-3.51-.81-6.72.24-7.31,2.46a3,3,0,0,0,.1,1.78c.6,1.72,2.55,3.37,5.16,4.18l.34.1.51.13.11,0h0a8.84,8.84,0,0,0-.84,3.79,8.74,8.74,0,0,0,4.71,7.8l-.23.14a3.12,3.12,0,0,0-.76,1.06,15.86,15.86,0,0, ...
+              <polygon points="415.35 438.25 417.53 438.69 424.96 449.68 421.69 450.12 415.35 438.25" fill="#2f2e41"></polygon>
+              <path d="M414.49,645.46a42.78,42.78,0,0,0,2.18,5.67,54,54,0,0,0,7.25-2.07,22.86,22.86,0,0,0-1.51-2,7.46,7.46,0,0,0-2.05-1.57,5.64,5.64,0,0,0-2.49,0C416.75,645.46,415.62,645.48,414.49,645.46Z" transform="translate(-52 -162.63)" fill="#fbbebe"></path>
+              <path d="M420.37,648.39a4.52,4.52,0,0,1-2.48-2.82A9.08,9.08,0,0,0,416,642.2a1.08,1.08,0,0,0-.84.63c-.18.32-.29.68-.49,1-.4.57-1.1.85-1.6,1.33s-.8,1.23-1.34,1.72a5.32,5.32,0,0,1-2.53,1,16.41,16.41,0,0,0-4,1.47,6.9,6.9,0,0,1-1.47.66,5.82,5.82,0,0,1-1.72,0l-3.94-.28a3.08,3.08,0,0,0-1,0,1.91,1.91,0,0,0-1.25,2,3,3,0,0,0,1.25,2.12,8.3,8.3,0,0,0,2.24,1.14,65.9,65.9,0,0,0,10.56,3v.78a18.28, ...
+              <path d="M495.93,633.54c-11.68.7-23.22-2.72-34.92-3.08a17.05,17.05,0,0,0-4.34.27,21.87,21.87,0,0,0-4.3,1.67c-6.27,2.92-12.76,5.34-19.24,7.75l-7.45,2.77a11.28,11.28,0,0,1-4.61,1c-.71,0-1.58-.15-2,.44-1.21,1.8-.64,4.43.16,6.45.26.65.7,1.36,1.4,1.43a4.07,4.07,0,0,0,1.42-.36c1.35-.36,2.7.55,4.1.7,1.72.18,3.32-.8,4.95-1.4,2.7-1,5.64-1,8.48-1.44,4.75-.79,9.16-3,13.28-5.51a15.5,15.5,0,0,1, ...
+              <path d="M495.93,633.54c-11.68.7-23.22-2.72-34.92-3.08a17.05,17.05,0,0,0-4.34.27,21.87,21.87,0,0,0-4.3,1.67c-6.27,2.92-12.76,5.34-19.24,7.75l-7.45,2.77a11.28,11.28,0,0,1-4.61,1c-.71,0-1.58-.15-2,.44-1.21,1.8-.64,4.43.16,6.45.26.65.7,1.36,1.4,1.43a4.07,4.07,0,0,0,1.42-.36c1.35-.36,2.7.55,4.1.7,1.72.18,3.32-.8,4.95-1.4,2.7-1,5.64-1,8.48-1.44,4.75-.79,9.16-3,13.28-5.51a15.5,15.5,0,0,1, ...
+              <path d="M436.92,646q2.74.93,5.53,1.72c1-1.71,3-2.9,3.43-4.86a.46.46,0,0,0,0-.26.48.48,0,0,0-.21-.19,10.48,10.48,0,0,0-4.34-1.27c-1.2-.07-1.75,1.18-2.29,2.06C438.83,643.59,437.41,646.19,436.92,646Z" transform="translate(-52 -162.63)" fill="#fbbebe"></path>
+              <path d="M439.48,644.78a4,4,0,0,1-2.69-1.77c-.25-.45-.53-1.06-1.05-1s-.69.6-.8,1.07a30,30,0,0,1-1.51,5.19,9.74,9.74,0,0,1-3.23,4.25c-.68.47-1.43.82-2.1,1.32a14.43,14.43,0,0,1-1.77,1.34,10.51,10.51,0,0,1-1.36.5,9.43,9.43,0,0,0-1.94,1,3.85,3.85,0,0,0-1.33,1.27,1.83,1.83,0,0,0-.1,1.77,1.89,1.89,0,0,0,1.29.88,5.44,5.44,0,0,0,1.61,0c2.13-.15,4.26-.34,6.39-.53a18.44,18.44,0,0,0,2.67-.37l1 ...
+              <path d="M478.42,614.8a9.12,9.12,0,0,1-4.66-4.62c-.19-.44-.35-.91-.54-1.35a10.26,10.26,0,0,0-2.78-3.87,1.37,1.37,0,0,0-1-.43,1.09,1.09,0,0,0-.75.81,3.09,3.09,0,0,0-.08,1.25,22.27,22.27,0,0,0,.5,2.92l.57,2.58a5.66,5.66,0,0,0,.35,1.15,4,4,0,0,0,.95,1.13c.52.48,1.05.94,1.58,1.4a5.43,5.43,0,0,0,1,.69,5.81,5.81,0,0,0,1.77.52,15.06,15.06,0,0,1,6.64,2.56s0-1.73,0-1.92a2.54,2.54,0,0,0-.5-1. ...
+              <path d="M478.42,614.8a9.12,9.12,0,0,1-4.66-4.62c-.19-.44-.35-.91-.54-1.35a10.26,10.26,0,0,0-2.78-3.87,1.37,1.37,0,0,0-1-.43,1.09,1.09,0,0,0-.75.81,3.09,3.09,0,0,0-.08,1.25,22.27,22.27,0,0,0,.5,2.92l.57,2.58a5.66,5.66,0,0,0,.35,1.15,4,4,0,0,0,.95,1.13c.52.48,1.05.94,1.58,1.4a5.43,5.43,0,0,0,1,.69,5.81,5.81,0,0,0,1.77.52,15.06,15.06,0,0,1,6.64,2.56s0-1.73,0-1.92a2.54,2.54,0,0,0-.5-1. ...
+              <ellipse cx="450.2" cy="429.57" rx="2.84" ry="2.86" fill="#2f2e41"></ellipse>
+              <path d="M500.44,590.48a3.07,3.07,0,0,0-2.24.32,3.11,3.11,0,0,0-.77,1.05,15.25,15.25,0,0,0-1.48,3.48c-.25,1-.29,2.13-.61,3.16-.58,1.9-2.07,3.51-2.24,5.49-.08.89.07,1.93-.58,2.53a3.68,3.68,0,0,1-.51.37,3.84,3.84,0,0,0-1.12,1.15,3.77,3.77,0,0,1-.79,1.09c-.58.43-1.54.39-1.82,1s.45,1.44.28,2.16-.8.86-1.25,1.26a4.37,4.37,0,0,0-1.08,2,6.67,6.67,0,0,1-5.44-1.35,17.32,17.32,0,0,1,.85,6.22,2 ...
+              <path d="M500.44,590.48a3.07,3.07,0,0,0-2.24.32,3.11,3.11,0,0,0-.77,1.05,15.25,15.25,0,0,0-1.48,3.48c-.25,1-.29,2.13-.61,3.16-.58,1.9-2.07,3.51-2.24,5.49-.08.89.07,1.93-.58,2.53a3.68,3.68,0,0,1-.51.37,3.84,3.84,0,0,0-1.12,1.15,3.77,3.77,0,0,1-.79,1.09c-.58.43-1.54.39-1.82,1s.45,1.44.28,2.16-.8.86-1.25,1.26a4.37,4.37,0,0,0-1.08,2,6.67,6.67,0,0,1-5.44-1.35,17.32,17.32,0,0,1,.85,6.22,2 ...
+              <path d="M497.78,636c-2.43-1.16-4.21-3.31-6.18-5.16-3.75-3.53-8.34-6-12.29-9.33-2.62-2.19-5-4.74-7.83-6.58a19.57,19.57,0,0,0-8.39-2.93,2.23,2.23,0,0,0-1,0,2.3,2.3,0,0,0-.95.75,30.22,30.22,0,0,0-3.39,4.65c-5,8.5-12.11,15.64-18,23.54a1.5,1.5,0,0,0-.29.56,1.39,1.39,0,0,0,.22.87l1.2,2.34c.67,1.32,1.75,2.85,3.19,2.59a3.49,3.49,0,0,0,1.67-1.07l5.38-5.15,5.64-5.39c3.3-3.15,6.66-6.38,8.82-1 ...
+              <ellipse cx="451.95" cy="431.55" rx="2.19" ry="2.2" fill="#e8e8f0"></ellipse>
+              <path d="M503.47,592.53a4.21,4.21,0,0,1,.24,3.16,12.82,12.82,0,0,1-1.41,2.91,13.85,13.85,0,0,1-1.83,2.6c3.38.16,6.65-1.16,9.62-2.76,2.4-1.29,4.79-2.88,6.07-5.29a2.8,2.8,0,0,1-2.24-.42,8,8,0,0,1-1.71-1.59l-1.69-1.9a7.61,7.61,0,0,1-1-1.37,3.78,3.78,0,0,1-.45-2.21,12.87,12.87,0,0,1-2.24,1.86c-1.21.79-2.54,1.36-3.76,2.14-.42.28-1.38.72-1.13,1.28S503.13,591.93,503.47,592.53Z" transform=" ...
+              <path d="M503.47,592.53a4.21,4.21,0,0,1,.24,3.16,12.82,12.82,0,0,1-1.41,2.91,13.85,13.85,0,0,1-1.83,2.6c3.38.16,6.65-1.16,9.62-2.76,2.4-1.29,4.79-2.88,6.07-5.29a2.8,2.8,0,0,1-2.24-.42,8,8,0,0,1-1.71-1.59l-1.69-1.9a7.61,7.61,0,0,1-1-1.37,3.78,3.78,0,0,1-.45-2.21,12.87,12.87,0,0,1-2.24,1.86c-1.21.79-2.54,1.36-3.76,2.14-.42.28-1.38.72-1.13,1.28S503.13,591.93,503.47,592.53Z" transform=" ...
+              <path d="M503.51,592.86a2.85,2.85,0,0,0,0,3.3c.1.12.46.87.61.88a.55.55,0,0,0,.38-.2,25.1,25.1,0,0,0,2.14-2.55,4.76,4.76,0,0,1,2.72-1.78,5.88,5.88,0,0,1,3.75,1.17,3.56,3.56,0,0,1,1.29,1,3.49,3.49,0,0,1,.42,1.43,68.9,68.9,0,0,1,.38,9.09,35.82,35.82,0,0,1-.43,6.55c-.22,1.11-.54,2.2-.69,3.33a21.24,21.24,0,0,0-.14,3v4.18a11,11,0,0,0,.16,2.29c.2.91.63,1.76.92,2.65a17.33,17.33,0,0,1,.61,3. ...
+              <path d="M492.36,625.5l1.33,0,.28-2a4.17,4.17,0,0,1,.28-1.17,7,7,0,0,1,.77-1.12c1.3-1.85,1.21-4.3,1.53-6.55a33.33,33.33,0,0,1,2.92-8.56l2.58-5.75a7.94,7.94,0,0,0,.56-1.47,8.82,8.82,0,0,0,.16-1.7c0-.89.28-4.13.31-5,0-.33-.73-.15-.88-.44-.32-.64-.76,1.09-1.37,1.47a3.36,3.36,0,0,0-1.19,1.74c-.56,1.44-.9,3-1.42,4.43a9.05,9.05,0,0,1-2.41,3.92c-.53.46-1.14.83-1.64,1.33-1.6,1.58-1.76,4.09- ...
+              <path d="M511.5,590.47a4.54,4.54,0,0,0-2.37,3.25,16.44,16.44,0,0,1-1.2,4c-1.62,2.82-5.75,3.64-6.9,6.69a12.3,12.3,0,0,0-.46,3.15c-.2,2.19-.92,4.3-1.14,6.48a28.87,28.87,0,0,0,.12,5.09L501,636.35a23,23,0,0,1-.43,7.58,7.3,7.3,0,0,0,4.34-.72c1.35-.62,2.59-1.47,3.9-2.19,2.24-1.24,4.72-2.11,6.69-3.76a4.09,4.09,0,0,0,.9-1A8.65,8.65,0,0,0,517,634a12.2,12.2,0,0,1,.65-1.78,52.45,52.45,0,0,0,1. ...
+              <path d="M481.29,618.57a10.43,10.43,0,0,1-2.75-2.09c-.36-.42-.66-.89-1-1.31a6.81,6.81,0,0,0-3.46-2,16.9,16.9,0,0,0-4-.39,7.26,7.26,0,0,0-2.67.32,2.72,2.72,0,0,0-1.77,1.9,3.15,3.15,0,0,0,1.35,2.8,6.36,6.36,0,0,0,2.69,1.3,21.5,21.5,0,0,0,2.43.29c4.78.5,9.05,3.1,13.24,5.47q5.52,3.14,11.26,5.84a23.4,23.4,0,0,0,.19-3.45c0-.86-.07-2.18-.88-2.69a17.66,17.66,0,0,0-3.58-1.07,56.34,56.34,0,0, ...
+              <ellipse cx="450.2" cy="420.35" rx="8.52" ry="8.57" fill="#fbbebe"></ellipse>
+              <path d="M513.93,600.18a13.66,13.66,0,0,0-1.65,3.09c-1,2.16-2.7,4-3.53,6.29s-.84,5.23-2.55,7.1a2.49,2.49,0,0,0-.75,1.08,5.16,5.16,0,0,1-.08.78c-.25.68-1.36.71-1.57,1.41-.06.2,0,.42-.1.62-.17.45-.77.52-1.16.79-.83.56-.66,1.86-1.25,2.67a2.25,2.25,0,0,1-2.13.75,7.5,7.5,0,0,1-2.21-.83,2.13,2.13,0,0,0-.8-.25,1.65,1.65,0,0,0-.92.32,5.67,5.67,0,0,0-2.3,4.85,15.84,15.84,0,0,0,1.32,5.43,2.18 ...
+              <path d="M531.05,589a9.6,9.6,0,0,0-3.94-7.8,11.41,11.41,0,0,0-9.69-11.11,11.07,11.07,0,0,0-21.81.59c-3.54-.8-6.78.24-7.37,2.42s1.91,5,5.64,6a10.51,10.51,0,0,0,3.62.36,11.08,11.08,0,0,0,7.62,4.75,11.37,11.37,0,0,0,8.32,8.58,9.55,9.55,0,0,0,2.57,3.55c0,.27,0,.55,0,.83a7.21,7.21,0,1,0,14.42,0,7.7,7.7,0,0,0-.69-3.19A9.64,9.64,0,0,0,531.05,589Z" transform="translate(-52 -162.63)" fill="# ...
+              <path d="M490.64,573.56c.59-2.18,3.83-3.22,7.37-2.42a11.26,11.26,0,0,1,11-9.48l.73,0a10.84,10.84,0,0,0-3.14-.47,11.26,11.26,0,0,0-11,9.48c-3.54-.8-6.78.24-7.37,2.42s1.91,5,5.64,6a11,11,0,0,0,2.14.37C492.44,578.39,490.05,575.79,490.64,573.56Z" transform="translate(-52 -162.63)" opacity="0.1"></path>
+              <path d="M518.37,597.59a7.52,7.52,0,0,1,0-.82,9.34,9.34,0,0,1-2.57-3.55,11.34,11.34,0,0,1-8.32-8.59,11.05,11.05,0,0,1-7.63-4.74,9.6,9.6,0,0,1-2.13-.06,11.11,11.11,0,0,0,7.35,4.37,11.37,11.37,0,0,0,8.32,8.58,9.55,9.55,0,0,0,2.57,3.55c0,.27,0,.55,0,.83a7.34,7.34,0,0,0,7.17,7.46A7.46,7.46,0,0,1,518.37,597.59Z" transform="translate(-52 -162.63)" opacity="0.1"></path>
+              <ellipse cx="505.04" cy="584.74" rx="1.31" ry="1.75" transform="translate(-302.82 261.95) rotate(-37.22)" fill="#fbbebe"></ellipse>
+</svg>
```
Let's move the images and illustrations to assets/images/filename

Iseluin commented on src/utils/utils.js:
```diff
@@ -0,0 +1,20 @@
+import emailjs from 'emailjs-com';
+
+const sendEmail = (values, modelRef) => {
+  emailjs
+    .send('service_va0wqwg', 'template_l4l91a5', values, 'JrpFs7BlCv1nsgafI')
+    .then(
+      () => {
+        modelRef.current.style.display = 'block';
+        modelRef.current.childNodes[0].style.display = 'block';
+        modelRef.current.childNodes[1].style.display = 'none';
+      },
+      () => {
+        modelRef.current.style.display = 'block';
+        modelRef.current.childNodes[1].style.display = 'block';
+        modelRef.current.childNodes[0].style.display = 'none';
+      },
+    );
+};
+
+export default sendEmail;
```
Great work in moving this function. One change could be renaming the file to something more intuitive. Like sendEmail.js.

Iseluin commented on src/pages/ContactUs.jsx:
```diff
@@ -0,0 +1,205 @@
+                      Full Name
+                      <input
+                        className="w-full bg-gray-200 text-gray-900 mt-2 p-3 rounded-lg focus:outline-none focus:shadow-outline"
+                        {...field}
```
These gives out eslint warnings like no-props-spreading. You can see the warnings in your console when you npm start. Can you find a workaround about it ?

androidmini9x replied:
I solved those errors

Iseluin reviewed (changes requested):
Great work @androidmini9x , some minor issues:

1- Errors and warnings related to the counterAPI and store, you can delete all related files to the redux and counterAPI since we wont be using them.
2- There are failed tests on contactus and app, we can delete app.test.js and write a test for it later. 
3- Let's switch the colors of the header and "Please get in touch..." parts. I think it will look better that way.

androidmini9x committed:
Fix: restructre the file & folder name

androidmini9x committed:
Fix: fix some eslint errors

Iseluin reviewed (changes requested):
Looks good. Only 1 issue. Please move the ContactUs.jsx and its tests inside src/pages/ContactUs folder.

androidmini9x committed:
Fix: Move ContactUs.jsx to its own folder

androidmini9x committed:
Fix: fix merged issues

androidmini9x committed:
Fix: renamed the navbar folder

androidmini9x committed:
Fix: fix merged issues

androidmini9x committed:
Fix: fix merged issues & eslint errors
